src/core/stdcpp/optional.d

```diff
@@ -0,0 +1,201 @@
+/**
+ * D header file for interaction with C++ std::optional.
+ *
+ * Copyright: Copyright (c) 2018 D Language Foundation
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Manu Evans
+ * Source:    $(DRUNTIMESRC core/stdcpp/optional.d)
+ */
+
+module core.stdcpp.optional;
+
+import core.stdcpp.exception : exception;
+
+version (CppRuntime_DigitalMars)
+{
+    pragma(msg, "std::optional not supported by DMC");
```

Comment:
I think this should be a generic "DMC does not support C++11 or newer" generated by the compiler. i.e. if one passes `-extern-std=c++1[1|4|7] -m32`.

Reply:
DMC does have some sparse C++11 features.

Reply:
Urgh, carry on then.